lib/posix/sleep.c

```diff
@@ -14,8 +14,12 @@
  */
 unsigned sleep(unsigned int seconds)
 {
-	k_sleep(K_SECONDS(seconds));
-	return 0;
+	int rem;
+
+	rem = k_sleep(K_SECONDS(seconds));
+	__ASSERT_NO_MSG(rem >= 0);
+
+	return rem / MSEC_PER_SEC;
```

Comment:
question: does the spec require rounding up? ie, if remaining is 500ms, should we return 0 or 1?

Reply:
It's unspecified 🥲

Reply:
But the return value of 0 must mean "the requested time has elapsed", no?

Reply:
Correct - `rem` must be non-negative